comp/core/hostname/component.go

```diff
@@ -6,20 +6,11 @@
 // Package hostname exposes hostname.Get() as a component.
 package hostname
 
-import (
-	"context"
-
-	"github.com/DataDog/datadog-agent/pkg/util/hostname"
-)
+import "github.com/DataDog/datadog-agent/comp/core/hostname/hostnameinterface"
 
 // team: agent-shared-components
 
 // Component is the component type.
 type Component interface {
-	// Get returns the host name for the agent.
-	Get(context.Context) (string, error)
-	// GetWithProvider returns the hostname for the Agent and the provider that was use to retrieve it.
-	GetWithProvider(ctx context.Context) (hostname.Data, error)
-	// GetSafe is Get(), but it returns 'unknown host' if anything goes wrong.
-	GetSafe(context.Context) string
+	hostnameinterface.HostnameInterface
```

Comment:
Create this as alias rather than an embeded type

Reply:
I get this [CI error](https://app.circleci.com/pipelines/github/DataDog/datadog-agent/115847/workflows/85184f49-d1a3-4953-a1f2-32d5f9b2d424/jobs/1365760) when I change it to an alias:
```
** comp/core/hostname/component.go does not define a Component interface; skipping
```

Reply:
This is getting fixed in PR https://github.com/DataDog/datadog-agent/pull/22095